rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -61,217 +72,564 @@ class LiaisonNegotiator : public rmf_traffic::schedule::Negotiator
+      != bid_notice_assignments.end())
+    return;
+
+  if (!accept_task)
+  {
+    RCLCPP_WARN(
+      node->get_logger(),
+      "Fleet [%s] is not configured to accept any task requests. Use "
+      "FleetUpdateHadndle::accept_task_requests(~) to define a callback "
+      "for accepting requests", name.c_str());
+
+    return;
+  }
+
+  if (!accept_task(msg->task_profile))
+  {
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Fleet [%s] is configured to not accept task [%s]",
+      name.c_str(),
+      msg->task_profile.task_id.c_str());
+
+      return;
+  }
+
+  if (!task_planner
+    || !initialized_task_planner)
+  {
+    RCLCPP_WARN(
+      node->get_logger(),
+      "Fleet [%s] is not configured with parameters for task planning."
+      "Use FleetUpdateHandle::set_task_planner_params(~) to set the "
+      "parameters required.", name.c_str());
 
-  const std::size_t n = request.num_loops;
-  if (n == 0)
-    return rmf_utils::nullopt;
+    return;
+  }
 
+  // Determine task type and convert to request pointer
+  rmf_task::ConstRequestPtr new_request = nullptr;
+  const auto& task_profile = msg->task_profile;
+  const auto& task_type = task_profile.task_type;
+  const rmf_traffic::Time start_time = rmf_traffic_ros2::convert(task_profile.start_time);
+  // TODO (YV) get rid of ID field in RequestPtr
+  std::string id = msg->task_profile.task_id;
   const auto& graph = planner->get_configuration().graph();
-  const auto loop_start_wp = graph.find_waypoint(request.start_name);
-  if (!loop_start_wp)
-    return rmf_utils::nullopt;
 
-  const auto loop_end_wp = graph.find_waypoint(request.finish_name);
-  if (!loop_end_wp)
-    return rmf_utils::nullopt;
+  // Process Cleaning task
+  if (task_type.type == rmf_task_msgs::msg::TaskType::TYPE_CLEAN)
+  {
+    if (task_profile.clean.start_waypoint.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [clean.start_waypoint] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    // Check for valid start waypoint
+    const std::string start_wp_name = task_profile.clean.start_waypoint;
+    const auto start_wp = graph.find_waypoint(start_wp_name);
+    if (!start_wp)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Fleet [%s] does not have a named waypoint [%s] configured in its "
+        "nav graph. Rejecting BidNotice with task_id:[%s]",
+        name.c_str(), start_wp_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    // Get dock parameters
+    const auto clean_param_it = dock_param_map.find(start_wp_name);
+    if (clean_param_it == dock_param_map.end())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Dock param for dock_name:[%s] unavailable. Rejecting BidNotice with "
+        "task_id:[%s]", start_wp_name.c_str(), id.c_str());
+
+      return;
+    }
+    const auto clean_param = clean_param_it->second;
```

Comment:
Nitpick: This could be `const auto&`.

Reply:
Done https://github.com/osrf/rmf_core/pull/217/commits/f73f6896aedf9ba33247fe43c9034138912a848a